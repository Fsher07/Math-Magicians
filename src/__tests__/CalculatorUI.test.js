import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Calculator from '../components/CalculatorUI';
import renderer from 'react-test-renderer';

describe('Calculator', () => {
  test('should render calculator', () => {
    render(<Calculator />);
    expect(screen.getByText('AC').value).toBe('AC');
  });
});

it('renders correctly', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});