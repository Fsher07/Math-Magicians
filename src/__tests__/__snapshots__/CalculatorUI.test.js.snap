// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="calculator-page"
>
  <div
    className="calculator-page-intro"
  >
    <h3
      className="calculator-title"
    >
      Let's do some math!
    </h3>
    <p>
      You can calculate the time it takes to achieve your dreams using the calculator on the left.
    </p>
  </div>
  <div
    className="calculator"
  >
    <div
      className="calculator-display-value"
    >
      0
    </div>
    <div
      className="calculator-keypad"
    >
      <div
        className="calculator-keypad-digit-column"
      >
        <button
          className="key AC"
          onClick={[Function]}
          type="button"
          value="AC"
        >
          AC
        </button>
        <button
          className="key sign"
          onClick={[Function]}
          type="button"
          value="+/-"
        >
          +/-
        </button>
        <button
          className="key percent"
          onClick={[Function]}
          type="button"
          value="%"
        >
          %
        </button>
        <button
          className="key digit"
          onClick={[Function]}
          type="button"
          value="7"
        >
          7
        </button>
        <button
          className="key digit"
          onClick={[Function]}
          type="button"
          value="8"
        >
          8
        </button>
        <button
          className="key digit"
          onClick={[Function]}
          type="button"
          value="9"
        >
          9
        </button>
        <button
          className="key digit"
          onClick={[Function]}
          type="button"
          value="4"
        >
          4
        </button>
        <button
          className="key digit"
          onClick={[Function]}
          type="button"
          value="5"
        >
          5
        </button>
        <button
          className="key digit"
          onClick={[Function]}
          type="button"
          value="6"
        >
          6
        </button>
        <button
          className="key digit"
          onClick={[Function]}
          type="button"
          value="1"
        >
          1
        </button>
        <button
          className="key digit"
          onClick={[Function]}
          type="button"
          value="2"
        >
          2
        </button>
        <button
          className="key digit"
          onClick={[Function]}
          type="button"
          value="3"
        >
          3
        </button>
        <button
          className="key zero"
          onClick={[Function]}
          type="button"
          value="0"
        >
          0
        </button>
        <button
          className="key decimal"
          onClick={[Function]}
          type="button"
          value="."
        >
          .
        </button>
      </div>
      <div
        className="calculator-keypad-function-column"
      >
        <button
          className="key divide bg-orange"
          onClick={[Function]}
          type="button"
          value="รท"
        >
          รท
        </button>
        <button
          className="key multiply bg-orange"
          onClick={[Function]}
          type="button"
          value="x"
        >
          ร
        </button>
        <button
          className="key subtract bg-orange"
          onClick={[Function]}
          type="button"
          value="-"
        >
          -
        </button>
        <button
          className="key add bg-orange"
          onClick={[Function]}
          type="button"
          value="+"
        >
          +
        </button>
        <button
          className="key equal bg-orange"
          onClick={[Function]}
          type="button"
          value="="
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
